bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,377 @@
+        file=sys.stderr,
+    )
+    this_all_keys_dict = {}
+
+    # Find S3 keys that are present in NZSL, or absent
+    for video_key in this_s3_bucket_raw_keys_list:
+        dict_row = this_nzsl_raw_keys_dict.get(video_key, None)
+        if dict_row:
+            # Split out for readability
+            [
+                gloss_idgloss,
+                gloss_created_at,
+                gloss_id,
+                video_id,
+                gloss_public,
+                video_public,
+            ] = dict_row
+
+            this_all_keys_dict[video_key] = [
+                True,  # NZSL PRESENT
+                True,  # S3 PRESENT
+                gloss_idgloss,
+                gloss_created_at,
+                gloss_id,
+                video_id,
+                gloss_public,
+                video_public,
+            ]
+        else:
+            this_all_keys_dict[video_key] = [
+                False,  # NZSL Absent
+                True,  # S3 PRESENT
+                "",  # gloss_idgloss,
+                "",  # gloss_created_at,
+                "",  # gloss_id
+                "",  # video_id,
+                "",  # gloss_public,
+                "",  # video_public,
+            ]
+
+    # Find NZSL keys that are absent from S3 (present handled above)
+    for video_key, [
+        gloss_idgloss,
+        gloss_created_at,
+        gloss_id,
+        video_id,
+        gloss_public,
+        video_public,
+    ] in this_nzsl_raw_keys_dict.items():
+        if video_key not in this_s3_bucket_raw_keys_list:
+            this_all_keys_dict[video_key] = [
+                True,  # NZSL PRESENT
+                False,  # S3 Absent
+                gloss_idgloss,
+                gloss_created_at,
+                gloss_id,
+                video_id,
+                gloss_public,
+                video_public,
+            ]
+
+    return this_all_keys_dict
+
+
+# Cases
+# In S3     In NZSL     Action
+#   Is        Not         Delete S3 Object
+#   Is        Is          Update ACL
+#   Not       Is          Review
+def get_action(key_in_nzsl, key_in_s3):
+    if key_in_s3:
+        if key_in_nzsl:
+            return "Update ACL"
+        else:
+            return "Delete S3 Object"
+    return "Review"
+
+
+# Get S3 object's ACL
+def get_s3_canned_acl(video_key):
+    result = aws_cli(
+        [
+            "s3api",
+            "get-object-acl",
+            "--output",
+            "text",
+            "--query",
+            "Grants[*].Permission",
+            "--bucket",
+            AWS_S3_BUCKET,
+            "--key",
+            video_key,
+        ]
+    )
+    acls_grants = result.stdout.strip().split("\t")
+
+    canned_acl = "unknown"
+    if len(acls_grants) > 1:
+        if acls_grants[0] == "FULL_CONTROL" and acls_grants[1] == "READ":
+            canned_acl = "public-read"
+    elif acls_grants[0] == "FULL_CONTROL":
+        canned_acl = "private"
+
+    return canned_acl
```

Comment:
suggestion: now that this is a function, we should be able to just `return` immediately rather than use a variable

Reply:
But then there's no default? 🤷🏻‍♂️ 
But Ok!

Reply:
Oh, yes there is, you mean inside the clauses.

Reply:
Done!